app/components/ui/SwapInput.tsx

```diff
@@ -37,7 +37,7 @@ export const SwapInput = ({
           className="w-full overflow-hidden overscroll-none bg-transparent text-3xl caret-text-primary-main outline-none placeholder:text-text-disabled"
           {...props}
         />
-        {selectedToken instanceof Token ? (
+        {selectedToken === null ? null : selectedToken instanceof Token ? (
```

Comment:
this looks a bit odd :/

Reply:
Yeah agree, maybe we could do something like abstracting this into a new component in this same file, and do an early return if no `selectedToken` is provided

```tsx
interface ComponentProps {
  selectedToken: Token | Outcome | null;
  onTokenClick?: (outcome?: Outcome) => void;
  tokenList: Array<Outcome> | null;
}

const Component = ({ selectedToken, tokenList, onTokenClick }: ComponentProps) => {
  if (!selectedToken) return null;

  return selectedToken instanceof Token ? (
    <Button className="flex-shrink-0" variant="pastel">
     /* ... */
    </Button>
  ) : (
    <Popover>
      /* ... */
    </Popover>
  );
};

```

Then inside the `SwapInput` we do: 
```tsx
  return (
    <div className="space-y-1 rounded-16 bg-surface-surface-1 p-4">
      <div className="text-xs font-semibold uppercase text-text-low-em">{title}</div>
      <div className="flex w-full space-x-2">
        <input
          ...
        />
        <Component
          selectedToken={selectedToken}
          tokenList={tokenList}
          onTokenClick={onTokenClick}
        />
      </div>
      {children}
    </div>
  );
```

Reply:
Found a simple solution for it. Let me know if it doesn't work.